qutip/core/cy/_element.pyx

```diff
@@ -263,7 +263,8 @@ cdef class _BaseElement:
     def __call__(self, t, args=None):
         if args:
             cache = []
-            self = self.replace_arguments(args, cache)
+            new = self.replace_arguments(args, cache)
+            return new.qobj(t) * new.coeff(t)
```

Comment:
Was this change needed? I'm fine with either version, just checking that I am not missing something and they are the same.